src/plugins/data/common/search/expressions/ip_prefix.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionDefinition, ExpressionValueBoxed } from '@kbn/expressions-plugin/common';
+
+export interface IpPrefix {
+  prefixLength?: number;
+  isIpv6?: boolean;
+}
+
+export type IpPrefixOutput = ExpressionValueBoxed<'ip_prefix', IpPrefix>;
+
+export type ExpressionFunctionIpPrefix = ExpressionFunctionDefinition<
+  'ipPrefix',
+  null,
+  IpPrefix,
+  IpPrefixOutput
+>;
+
+export const ipPrefixFunction: ExpressionFunctionIpPrefix = {
+  name: 'ipPrefix',
+  type: 'ip_prefix',
+  inputTypes: ['null'],
+  help: i18n.translate('data.search.functions.ipPrefix.help', {
+    defaultMessage: 'Create extended bounds',
```

Comment:
These help messages seem wrong to me

Reply:
Whoops, thank you for catching that, fixed to use language from https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-ipprefix-aggregation.html#search-aggregations-bucket-ipprefix-aggregation